crates/editor/src/editor.rs

```diff
@@ -10088,17 +10120,40 @@ impl Editor {
                         group_end = entry.range.end;
                     }
                     if entry.diagnostic.is_primary {
-                        primary_range = Some(entry.range.clone());
-                        primary_message = Some(entry.diagnostic.message.clone());
+                        primary = Some((entry.range.clone(), entry.diagnostic.message.clone()));
                     }
                     Some(entry)
                 })
                 .collect::<Vec<_>>();
-            let primary_range = primary_range?;
-            let primary_message = primary_message?;
+
+            if primary.is_none() {
+                if let Some(active_group) = self.active_diagnostics.remove(&group_id) {
+                    display_map.remove_blocks(active_group.blocks.into_keys().collect(), cx);
+                }
+                return;
+            }
+
+            let (primary_range, primary_message) = primary.unwrap();
```

Comment:
> .unwrap()

Please rewrite this as `match primary` or any other way that does not introduce another potential panic.
At worst, an `.expect("message")` has to be used instead of `.unwrap()` but it's not the case here.

Reply:
Generally curious here to further my Rust knowledge. Looking at this code I certainly see ways to write it better today, but given that is_none/return a few lines above, this unwrap is obviously not going to panic.

The obvious counter argument is "what's true today isn't necessary necessarily true tomorrow" after a separate PR changes the logic and leaves a hole where the unwrap could trigger. Which, fair point! Especially on a project as large as Zed.

I just haven't (yet?) seen a "Thou shalt not unwrap!" commandment adopted anywhere which I'm starting to entertain as a possibility for a core library I've been working on. Speaking of, does Zed have a lint for this?